src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,45 +1,67 @@
 import React, { useState } from 'react';
-import { useSelect } from 'downshift';
+import { useSelect, PropGetters } from 'downshift';
 import { usePopper } from 'react-popper';
 import { bem } from '../../utils/bem/bem';
 import { mergeRefs } from '../../utils/mergeRefs';
 import { PopupPlacement } from '../../constants';
 import { ButtonProps } from '../Buttons';
-import { ListItemProps } from '../List/ListItem';
 import { List } from '../List';
 import styles from './Dropdown.scss';
 
 const { elem } = bem('Dropdown', styles);
 
+export type DropdownRenderArgs<V> = {
+    getItemPropsByIndex: (index: number) => PropGetters<V>;
+};
+
 interface Props<V> extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
-    /**
-     *  Popup content to be placed inside List.
-     *  Navigation available only if element has not empty `value` attribute
-     *  and empty/false `disabled` attribute.
-     *  Use {@link ListItem} component.
-     */
-    children:
-        | React.ReactElement<ListItemProps>
-        | React.ReactElement<ListItemProps>[]
-        | (React.ReactElement<ListItemProps> | React.ReactElement<ListItemProps>[])[];
+    /** List of items */
+    items: V[];
+    /** Popup content */
+    children: (args: DropdownRenderArgs<V>) => React.ReactElement;
     /** {@link Button} element, controlled by current component */
     button: React.FunctionComponentElement<ButtonProps>;
+    /** onChange called when an item is selected */
+    onChange: (value: V) => void;
+    /** onOpen called on request to be open */
+    onOpen?: () => void;
+    /** onClose called on request to be close */
+    onClose?: () => void;
+    /** Should the children be shown. This prop is used when we need to have Dropdown as "controlled" component */
+    isOpen?: boolean;
     /**
-     * Callback called on selecting one of the passed as children items.
-     * Value parameter it is a `value` attribute of children item ({@link ListItemProps.value}).
+     * Convert item to string.
+     * Prop is required if passed items are not list of strings.
+     * This is need for accessibility aria-live messages for downshift.
      */
-    onChange: (value: V) => void;
+    itemToString?: (item: V | null) => string;
```

Comment:
This prop confuses me.
We have to implement list rendering on our own, but also send items list and this prop... Why do we need it if do everything on our own?